src/LibDeclarations/NativeExtensions/FeaturePoints.md

```diff
@@ -0,0 +1,61 @@
+# Feature point cloud
+## Introduction
+This document proposes an API for exposing Feature Point Clouds for WebXR.
+
+## Overview
+The API is synchronous, and on demand.  When served a new XRFrame the consuming app may query for the current feature point cloud for a given frame. As feature points comprise a large set of raw data it is best not to expose this to the JS application unless needed for app processing, so feature points must be requested handled in an on-demand fashion in order to reduce the amount of data presented to the consuming app, which must set its own frequency for updating its world understanding.
+
+The returned array will contain the feature point cloud tracked in the current frame, including points that may not be currently visible.  Feature points will be presented in a flat array of numbers where each point is presented as a set of 5 entries representing the X, Y, Z position in world space, the confidence value, and an ID that is durable across frames.

```

Comment:
```Feature points will be presented in a flat array of numbers where each point is presented as a set of 5 entries representing the X, Y, Z position in world space, the confidence value, and an ID that is durable across frames.
```

The above is slightly confusing -- I guess part of it is my lack of deep knowledge between `XRFrame` vs. `WebXRFeaturePointSystem`. I understand that client would be using `WebXRFeaturePointSystem` and will be working with array of IWebXRFeaturePoint. Is XFrame just implementation details then?

Reply:
From the point of view of the consumer of BabylonJS yes the XRFrame is an implementation detail.  Basically the entire WebXR surface area is abstracted by Babylonjs and presented through various modules that the user attaches and observes.

In this case the WebXRFeaturePointSystem is an abstraction on the XRFrame's newly exposed featurePointCloud API.